packages/block-library/src/navigation-link/hooks.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * WordPress dependencies
+ */
+import { addFilter } from '@wordpress/hooks';
+import {
+	category,
+	page,
+	postTitle,
+	tag,
+	customPostType,
+} from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import fallbackVariations from './fallback-variations';
+
+function getIcon( variationName ) {
+	switch ( variationName ) {
+		case 'post':
+			return postTitle;
+		case 'page':
+			return page;
+		case 'tag':
+			return tag;
+		case 'category':
+			return category;
+		default:
+			return customPostType;
+	}
+}
+
+function enhanceNavigationLinkVariations( settings, name ) {
+	if ( name !== 'core/navigation-link' ) {
+		return settings;
+	}
+
+	// Fallback handling may be deleted after supported WP ranges understand the `variations`
+	// property when passed to register_block_type_from_metadata in index.php
+	if ( ! settings.variations ) {
+		return {
+			...settings,
+			variations: [ ...fallbackVariations ],
```

Comment:
I don't think we'd be altering stuff unintentionally by modifying `settings.variations` beyond this point, so cloning the array might not be necessary; is there another reason for it? I'll leave a suggestion up in case there isn't 😄 

```suggestion
			variations: fallbackVariations,
```

Reply:
Mainly it's to guard against array mutations (esp if other folks hook into `blocks.registerBlockType`), but since we can probably assume we only load core blocks once this is probably okay to drop. Removed in 5247f5666806f7f393ae9619436740dcdee5337b